File: src/main/java/org/luxons/sevenwonders/game/effects/InstantOwnBoardEffect.java
package org.luxons.sevenwonders.game.effects;

import org.luxons.sevenwonders.game.api.Table;
import org.luxons.sevenwonders.game.boards.Board;

public abstract class InstantOwnBoardEffect implements Effect {

    @Override
    public void apply(Table table, int playerIndex) {
        apply(table.getBoard(playerIndex));
    }

    protected abstract void apply(Board board);

    @Override
    public int computePoints(Table table, int playerIndex) {
        // InstantEffects are only important when applied to the board, they don't give extra points in the end
        return 0;
    }
}

File: game-engine/src/main/kotlin/org/luxons/sevenwonders/game/effects/InstantOwnBoardEffect.kt
package org.luxons.sevenwonders.game.effects

import org.luxons.sevenwonders.game.Player
import org.luxons.sevenwonders.game.boards.Board

abstract class InstantOwnBoardEffect : Effect {

    override fun applyTo(player: Player) = applyTo(player.board)

    protected abstract fun applyTo(board: Board)

    // InstantEffects are only important when applied to the board, they don't give extra points in the end
    override fun computePoints(player: Player): Int = 0
}
